cmake/RippledCore.cmake

```diff
@@ -130,33 +131,48 @@ target_link_modules(xrpl PUBLIC
 #     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
 #     $<INSTALL_INTERFACE:include>)
 
-target_compile_definitions(xrpl.libxrpl
-  PUBLIC
-    BOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT
-    BOOST_CONTAINER_FWD_BAD_DEQUE
-    HAS_UNCAUGHT_EXCEPTIONS=1)
-
-target_compile_options(xrpl.libxrpl
-  PUBLIC
-    $<$<BOOL:${is_gcc}>:-Wno-maybe-uninitialized>
-    $<$<BOOL:${voidstar}>:-DENABLE_VOIDSTAR>
+# These settings must be applied to all the header modules, too.
+set(base "xrpl.libxrpl")
+foreach(submodule
+  ""
+  ${submodules}
 )
+  STRING(LENGTH "${submodule}" len)
+  if(len EQUAL 0)
+    set(module "${base}")
+  else()
+    set(module "${base}.${submodule}")
+  endif()
+  message(STATUS "Setting up module ${module}")
+
+  target_compile_definitions("${module}"
+    PUBLIC
+      BOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT
+      BOOST_CONTAINER_FWD_BAD_DEQUE
+      HAS_UNCAUGHT_EXCEPTIONS=1)
+
+  target_compile_options("${module}"
+    PUBLIC
+      $<$<BOOL:${is_gcc}>:-Wno-maybe-uninitialized>
+      $<$<BOOL:${voidstar}>:-DENABLE_VOIDSTAR>
+  )
 
-target_link_libraries(xrpl.libxrpl
-  PUBLIC
-    LibArchive::LibArchive
-    OpenSSL::Crypto
-    Ripple::boost
-    Ripple::opts
-    Ripple::syslibs
-    absl::random_random
-    date::date
-    ed25519::ed25519
-    secp256k1::secp256k1
-    xrpl.libpb
-    xxHash::xxhash
-    $<$<BOOL:${voidstar}>:antithesis-sdk-cpp>
-)
+  target_link_libraries("${module}"
+    PUBLIC
+      LibArchive::LibArchive
+      OpenSSL::Crypto
+      Ripple::boost
+      Ripple::opts
+      Ripple::syslibs
+      absl::random_random
+      date::date
+      ed25519::ed25519
+      secp256k1::secp256k1
+      xrpl.libpb
+      xxHash::xxhash
+      $<$<BOOL:${voidstar}>:antithesis-sdk-cpp>
+  )
+endforeach()
```

Comment:
Instead of doing this for each submodule, you can do it once for the `INTERFACE` library further up this file, `xrpl.imports.main`, which is linked by all the submodules (directly or indirectly). In fact, on line 52, `xrpl.imports.main` is already linked to exactly this list (except `antithesis-sdk-cpp`). All you need to add to `xrpl.imports.main` is the preprocessor definitions, compiler options, and Antithesis.